pkg/statement/statement.go

```diff
@@ -172,15 +172,16 @@ func (a *AbstractStatement) AlgorithmInplaceConsideredSafe() error {
 			if spec.NewColumns[0].Tp != nil && spec.NewColumns[0].Tp.GetType() == mysql.TypeVarchar {
 				continue
 			}
+			unsafeClauses++
```

Comment:
Was this a bug? It might be, I'm not quite understanding from the diff if we missed a branch where we were supposed to increment `unsafeClauses`.

Reply:
I read through it, looks like a bug. Resolving this.